build.sbt

```diff
@@ -91,29 +88,49 @@ libraryDependencies ++= Seq(
   "org.scalamock" %% "scalamock" % "5.2.0" % Test
 )
 
+// spark >= 3.3 uses log4j 2.x while previous version relied on log4j 1.x
+libraryDependencies ++= {
+  if (testSparkVersion.value >= "3.3.0") {
+    Seq("org.apache.logging.log4j" % "log4j-core" % "2.19.0" % Test)
+  } else {
+    Seq("org.slf4j" % "slf4j-api" % "1.7.36" % "provided")
+  }
+}
+
 // Custom source layout for Spark Data Source API 2
 Compile / unmanagedSourceDirectories := {
-  if (testSparkVersion.value >= "3.2.2") {
+  if (testSparkVersion.value >= "3.3.0") {
     Seq(
       (Compile / sourceDirectory)(_ / "scala"),
-      (Compile / sourceDirectory)(_ / "3.x/scala"),
-      (Compile / sourceDirectory)(_ / "3.1_3.2/scala"),
-      (Compile / sourceDirectory)(_ / "3.2/scala")
+      (Compile / sourceDirectory)(_ / "3.3/scala"),
+      (Compile / sourceDirectory)(_ / "3.0_and_up/scala"),
+      (Compile / sourceDirectory)(_ / "3.1_and_up/scala"),
+      (Compile / sourceDirectory)(_ / "3.2_and_up/scala")
+    ).join.value
+  } else if (testSparkVersion.value >= "3.2.0") {
```

Comment:
Is the change from `3.2.2` to `3.2.0` intended here? Not sure why it was `3.2.2` in the first place, just making sure :)

Reply:
Yes, this was intentional. It broke the pattern that we check on minor version and not on patch level